api/buckets/buckets.go

```diff
@@ -22,20 +37,56 @@ var (
 )
 
 type Bucket struct {
-	Key       string `json:"_id"`
-	Name      string `json:"name"`
-	Path      string `json:"path"`
-	DNSRecord string `json:"dns_record,omitempty"`
-	CreatedAt int64  `json:"created_at"`
-	UpdatedAt int64  `json:"updated_at"`
+	Key       string   `json:"_id"`
+	Name      string   `json:"name"`
+	Path      string   `json:"path"`
+	DNSRecord string   `json:"dns_record,omitempty"`
+	Archives  Archives `json:"archives"`
+	CreatedAt int64    `json:"created_at"`
+	UpdatedAt int64    `json:"updated_at"`
+}
+
+type Archives struct {
+	Current Archive   `json:"current"`
+	History []Archive `json:"history"`
+}
+
+type Archive struct {
+	Cid   string `json:"cid"`
+	Deals []Deal `json:"deals"`
+}
+
+type Deal struct {
+	ProposalCid string `json:"proposalCid"`
+	Miner       string `json:"miner"`
 }
 
 func init() {
-	schema = jsonschema.Reflect(&Bucket{})
+	reflector := jsonschema.Reflector{ExpandedStruct: true}
+	schema = reflector.Reflect(&Bucket{})
 }
 
 type Buckets struct {
-	Threads *dbc.Client
+	ffsCol   *collections.FFSInstances
+	threads  *dbc.Client
+	pgClient *powc.Client
+
+	lock   sync.Mutex
+	ctx    context.Context
+	cancel context.CancelFunc
+	wg     sync.WaitGroup
+}
+
+func New(t *dbc.Client, pgc *powc.Client, col *collections.FFSInstances) *Buckets {
```

Comment:
Preferred to create a constructor here to control a bit how a `Buckets` should be constructed since got a bit more complex.

Reply:
Took advantage of that and made attributes unexported to be a bit safer from external changes.